ipv8/REST/health_endpoint.py

```diff
@@ -0,0 +1,112 @@
+    """
+    This endpoint manages measurements of non-functional requirements.
+    """
+
+    def __init__(self):
+        super(HealthEndpoint, self).__init__()
+        self.strategy = None
+        self.enabled = False
+
+    def setup_routes(self):
+        self.app.add_routes([web.get('/drift', self.retrieve_drift), web.put('/enable', self.enable_measurements)])
+
+    def enable(self):
+        if not self.session:
+            return False
+        if not self.enabled:
+            self.strategy = DriftMeasurementStrategy(self.session.walk_interval)
+            with self.session.overlay_lock:
+                self.session.strategies.append((self.strategy, -1))
+            return True
+        return True
+
+    def disable(self):
+        if not self.session:
+            return False
+        if self.enabled:
+            with self.session.overlay_lock:
+                self.session.strategies = [s for s in self.session.strategies
+                                           if not isinstance(s[0], DriftMeasurementStrategy)]
+            return True
+        return True
+
+    @docs(
+        tags=["Health"],
+        summary="Measure the core drift.",
+        responses={
+            200: {
+                "schema": schema(DriftResponse={"measurements": [
+                    schema(Measurement={
+                        "timestamp": Float,
+                        "drift": Float
+                    })
+                ]})
+            },
+            400: {
+                "schema": DefaultResponseSchema,
+                "examples": {'': {"success": False, "error": "Core drift disabled."}}
+            }
+        }
+    )
+    async def retrieve_drift(self, _):
+        if self.strategy is None:
+            return Response({"success": False, "error": "Core drift disabled."}, status=HTTP_NOT_FOUND)
+        with self.strategy.walk_lock:
+            return Response({"measurements": [{"timestamp": e[0], "drift": e[1]} for e in self.strategy.history]})
+
+    @docs(
+        tags=["Health"],
+        summary="Enable or disable measurements.",
+        parameters=[{
+            'in': 'body',
+            'name': 'enable',
+            'description': 'Whether to enable or disable measuring.',
+            'type': 'boolean',
+            'required': True
+        }],
+        responses={
+            200: {
+                "schema": DefaultResponseSchema,
+                "examples": {'true': {"success": True, "error": ""}}
+            },
+            400: {
+                "schema": DefaultResponseSchema,
+                "examples": {'Incorrect parameters': {"success": False, "error": "incorrect parameters"}}
+            }
+        }
+    )
+    async def enable_measurements(self, request):
+        enabled = await request.json()
+        status = self.enable() if enabled else self.disable()
+        return Response({"success": status, "error": "" if status else "Session not initialized."})
```

Comment:
Maybe drop the `error` field if there is no error? The `DefaultResponseSchema` already allows for this.